Source/NSLayoutConstraint.m

```diff
@@ -0,0 +1,601 @@
+  if ([coder allowsKeyedCoding])
+    {
+      [coder encodeFloat: _constant
+                  forKey: @"NSConstant"];
+      [coder encodeInteger: _firstAttribute
+                    forKey: @"NSFirstAttribute"];
+      [coder encodeObject: _firstItem
+                   forKey: @"NSFirstItem"];
+      [coder encodeInteger: _secondAttribute
+                    forKey: @"NSSecondAttribute"];
+      [coder encodeObject: _secondItem
+                   forKey: @"NSSecondItem"];
+      [coder encodeFloat: _multiplier
+                    forKey: @"NSMultiplier"];
+      [coder encodeInteger: _relation
+                    forKey: @"NSRelation"];
+      [coder encodeFloat: _priority
+                    forKey: @"NSPriority"];
+    }
+  else
+    {
+      [coder encodeValueOfObjCType: @encode(float)
+                                at: &_constant];
+      [coder encodeValueOfObjCType: @encode(NSUInteger)
+                                at: &_firstAttribute];
+      [coder encodeObject: _firstItem];
+      [coder encodeValueOfObjCType: @encode(NSUInteger)
+                                at: &_secondAttribute];
+      [coder encodeObject: _secondItem];       
+      [coder encodeValueOfObjCType: @encode(float)
+                                at: &_multiplier];
+      [coder encodeValueOfObjCType: @encode(NSUInteger)
+                                at: &_relation];
+      [coder encodeValueOfObjCType: @encode(float)
+                                at: &_priority];
+    }
+}
+
+- (id) copyWithZone: (NSZone *)z
+{
+  NSLayoutConstraint *constraint = [[NSLayoutConstraint allocWithZone: z]
+                                     initWithItem: _firstItem
+                                        attribute: _firstAttribute
+                                        relatedBy: _relation
+                                           toItem: _secondItem
+                                        attribute: _secondAttribute
+                                       multiplier: _multiplier
+                                         constant: _constant
+                                         priority: _priority];
+  return constraint;
+}
+
+- (void) dealloc
+{
+  [NSLayoutConstraint _removeConstraint: self];
+  [super dealloc];
+}
+
+- (NSString *) description
+{
+  return [NSString stringWithFormat: @"%@ <firstItem = %@, firstAttribute = %ld, relation = %ld, secondItem = %@, "
+                   "secondAttribute = %ld, multiplier = %f, constant = %f, priority = %f>",
+                   [super description],
+                   _firstItem,
+                   _firstAttribute,
+                   _relation,
+                   _secondItem,
+                   _secondAttribute,
+                   _multiplier,
+                   _constant,
+                   _priority];
+}
+
+// item1.attribute1 = multiplier × item2.attribute2 + constant
+- (void) _applyConstraint
+{
+  // Currently not implemented.
+}
+
++ (void) _handleWindowResize: (NSNotification *)notification
```

Comment:
Please remove this method and the notification handling for this, as this isn't the way we should try to implement constraint handling.

Reply:
Done